formats/json/jvmMain/src/kotlinx/serialization/json/internal/CharsetReader.kt

```diff
@@ -20,7 +20,7 @@ internal class CharsetReader(
             .onMalformedInput(CodingErrorAction.REPLACE)
             .onUnmappableCharacter(CodingErrorAction.REPLACE)
         byteBuffer = ByteBuffer.wrap(ByteArrayPool8k.take())
-        byteBuffer.flip() // Make empty
```

Comment:
Please additionally write in the comments why an explicit cast is needed here.

Problem with the return type.
In `Buffer`, the `flip` method was final, so it returned `Buffer'.

In subsequent versions of Java, the method was opened, and was overridden for `ByteBuffer`, thereby it began to return `ByteBuffer`

You can see this if you look at what javac compiles `byteBuffer.flip();`
for java 8 `INVOKEVIRTUAL java/nio/ByteBuffer.flip ()Ljava/nio/Buffer;`
for java 11 `INVOKEVIRTUAL java/nio/ByteBuffer.flip ()Ljava/nio/ByteBuffer;`

Reply:
Done. A comment added.